### DISCUSSION TITLE
Yes and no questions about what is possible in Geany 2.0

### DISCUSSION THREAD

**Discussion by ineuw** [geany/geany#3807] [General]
Geany is my only text editor in Linux Mint Cinnamon 21.3. Because of this, some of my questions may also depend on GTK, and not just Geany. My needs are editing text, prose, etc., and not programming related, which makes some of these questions strange.

Can I search and replace in all open documents? 

Can I add the color picker to the context menu?

Can I pin a list of files to the menu bar for direct access? This is not related to the recently accessed list, nor can be directly accessed from the default folder. Nor are they templates.

Can I increase tab width to a user defined number of columns?

Can templates be created with preset features, like tab widths? These would be indicated by the filename extensions like .t24, .t32, etc.

Can I disable matching highlight for all enclosures, brackets, braces, errors, quotation marks etc?

**Comment by elextr** [comment]
> Can I search and replace in all open documents?

Yes, choose "In Session" on the replace dialog.

> Can I add the color picker to the context menu?

No[^2]

> Can I pin a list of files to the menu bar for direct access? This is not related to the recently accessed list, nor can be directly accessed from the default folder. Nor are they templates.

No[^1]

> Can I increase tab width to a user defined number of columns?

Yes up to 8, per file `Document->Indent width`, per project `Project->Properties->Indentation->Width`, per filetype in the filetype file, default `Edit->Preferences->Editor->Indentation->Width` first found in this list wins.

> Can templates be created with preset features, like tab widths? These would be indicated by the filename extensions like .t24, .t32, etc.

Templates with preset contents can be defined, see the manual, but cannot define file specific settings for them [^1]

> Can I disable matching highlight for all enclosures, brackets, braces, errors, quotation marks etc?

No[^2] but can be minimised by adjusting styling see #3555

[^1]: could be done in a plugin, but "somebody" has to write it
[^2]: needs to be coded, if the capability exists it may only need an option to enable/disable it, but "somebody" has to code it

**Comment by elextr** [comment]
> Geany is my only text editor in Linux Mint Cinnamon 21.3.

I havn't upgraded to 21.3 yet, but ...

Are you sure? Normally Xed is provided as the default text editor under the menu item "Accessories->Text Editor" and of course you can install others from the software centre.

I'm not encouraging you to move away from Geany, but if it doesn't meet some of your needs its important that you check out the others and see if they are any better.  I suspect each one will do a different subset of the things you want ;-)

**Reply by ineuw** [reply]
I purged xed which automatically made Geany the default text editor. xed is anemic.
